src/frame.ts

```diff
@@ -660,11 +664,12 @@ export const getTargetFrame = (
     : getContainingFrame(_element);
 };
 
+// TODO: this a huge bottleneck for large scenes, optimise
```

Comment:
Frames should be re-checked and optimised further, currently, this is one of the biggest bottlenecks.